pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiClient.java

```diff
@@ -72,38 +84,50 @@ long insertClient(@Bind("clientName")String clientName,
     @SqlUpdate("DELETE FROM "
             + "     client "
             + "WHERE "
-            + "     client.client_name = :clientName"
+            + "     auth0_client_id = :auth0ClientId AND auth0_tenant_id = :auth0TenantId"
     )
-    int deleteClientByName(@Bind("clientName") String clientName);
+    int deleteClientByAuth0ClientIdAndAuth0TenantId(
+            @Bind("auth0ClientId") String auth0ClientId,
+            @Bind("auth0TenantId") long auth0TenantId
+    );
 
     @SqlQuery("SELECT "
             + "     client_id "
             + "FROM "
             + "     client "
             + "WHERE "
-            + "     client.client_name = :clientName"
+            + "     auth0_client_id = :auth0ClientId AND auth0_tenant_id = :auth0TenantId"
     )
-    Optional<Long> getClientIdByName(@Bind("clientName") String clientName);
-
-    @SqlQuery("SELECT client_id,client_name,auth0_client_id,auth0_signing_secret,web_password_redirect_url,is_revoked,auth0_tenant_id"
-            + "  FROM client WHERE auth0_client_id = :auth0ClientId")
-    @RegisterConstructorMapper(ClientDto.class)
-    Optional<ClientDto> findByAuth0ClientId(@Bind("auth0ClientId") String auth0ClientId);
+    Optional<Long> getClientIdByAuth0ClientIdAndAuth0TenantId(
+            @Bind("auth0ClientId") String auth0ClientId,
+            @Bind("auth0TenantId") long auth0TenantId
+    );
 
-    @SqlQuery("SELECT client_id,client_name,auth0_client_id,auth0_signing_secret,web_password_redirect_url,is_revoked,auth0_tenant_id"
-            + "  FROM client WHERE client_name= :clientName")
+    @SqlQuery("SELECT client_id,auth0_client_id,auth0_signing_secret,web_password_redirect_url,is_revoked,auth0_tenant_id"
+            + "  FROM client WHERE auth0_client_id = :auth0ClientId AND auth0_tenant_id = :auth0TenantId")
     @RegisterConstructorMapper(ClientDto.class)
-    Optional<ClientDto> findByClientName(@Bind("clientName") String clientName);
+    Optional<ClientDto> findByAuth0ClientIdAndAuth0TenantId(
+            @Bind("auth0ClientId") String auth0ClientId,
+            @Bind("auth0TenantId") long auth0TenantId
+    );
 
-    @SqlUpdate("UPDATE client SET web_password_redirect_url = :url WHERE auth0_client_id = :auth0ClientId")
-    int updateWebPasswordRedirectUrlByAuth0ClientId(
+    @SqlUpdate(
+            "UPDATE client SET web_password_redirect_url = :url WHERE auth0_client_id = :auth0ClientId "
+            + "AND EXISTS (SELECT 1 FROM auth0_tenant WHERE auth0_domain = :auth0Domain)"
```

Comment:
This checks that the auth0 domain exists but I don't see how it guarantees that it's the one we're looking for. Shouldn't we use `update ... join ... set` syntax here? Likewise for the SQL below.

Reply:
Done.